benchmarks/swe_bench/swe_bench.py

```diff
@@ -0,0 +1,372 @@
+    Plan,
+    TaskState,
+    generate,
+    system_message,
+    use_tools,
+)
+from inspect_ai.tool import bash
+from inspect_ai.util import sandbox
+
+INPUT_PROMPT = "Please solve the following issue:\n\n{issue_text}"
+COMPOSE_FILE_DIR = Path(__file__).parent / "resources/compose_files/"
+os.makedirs(COMPOSE_FILE_DIR, exist_ok=True)
+
+SAMPLE_TO_IMAGE_PATH = COMPOSE_FILE_DIR / "sample_to_image.json"
+
+DEFAULT_PLAN = Plan(
+    [
+        system_message("Please solve the task below."),
+        use_tools([bash()]),
+        generate(),
+    ],
+    name="simple_bash_plan",
+)
+
+DEFAULT_MAX_MESSAGES = 10
+
+
+@task
+def swe_bench(
+    dataset: str = "princeton-nlp/SWE-bench_Verified",
+    split: str = "test",
+    plan: Plan = DEFAULT_PLAN,
+    max_messages: int = DEFAULT_MAX_MESSAGES,
+    filter: Callable[[Sample], bool] | None = None,
+) -> Task:
+    """Returns a Task, representing an evaluation on SWE-bench.
+
+    Args.
+        dataset : str
+            The dataset to use. This should  either be the name of a dataset in the HF hub, or a path to a dataset on disk.
+        split : str
+            The split of the dataset to load.
+        plan : Plan
+            The plan to use when creating the task. If None, uses the default pkan.
+        max_messages : int
+            The maximum number of messages to generate for each sample.
+        filter : Callable[[Sample],bool]
+            A function to filter whether specific SWE-bench samples are included.
+    """
+    samples = hf_dataset(
+        dataset,
+        split=split,
+        sample_fields=FieldSpec(
+            input="problem_statement",
+            id="instance_id",
+            metadata=[
+                "base_commit",
+                "patch",
+                "PASS_TO_PASS",
+                "test_patch",
+                "version",
+                "repo",
+                "environment_setup_commit",
+            ],
+        ),
+    )
+
+    if filter:
+        samples = samples.filter(filter)
+
+    for sample in samples:
+        sample.input = INPUT_PROMPT.format(issue_text=sample.input)
+        sample.sandbox = (
+            "docker",
+            get_compose_file(sample.metadata["environment_setup_commit"], sample.id),
+        )
+        sample.setup = get_setup_script(
+            sample.metadata["repo"],
+            sample.metadata["version"],
+            sample.metadata["base_commit"],
+        )
+
+    return Task(
+        name=f"{dataset}_{plan.name}",
+        dataset=samples,
+        plan=plan,
+        scorer=swebench_scorer(),
+        max_messages=max_messages,
+    )
+
+
+def get_setup_script(repo: str, version: str, base_commit: str) -> str:
+    """Create a list of bash commands to set up the repository for testing. These are ran at the start of the sample,  clone the repository, and do some extra repository-specific installation steps over and above what is in the environment images."""
+    setup_script = f"""#!/bin/bash
```

Comment:
To make this flow a bit nicer you can indent the """ content and then call `textwrap.dedent` on it, e.g. see https://github.com/UKGovernmentBEIS/inspect_ai/blob/main/examples/agents/intercode-ctf/task.py